### PR TITLE
Fixing Race Condition incase of Screenshot Cmd

### PR DESCRIPTION
**Pull request by Amit3200**
- Fixing Race Condition incase of Screenshot Cmd
- Currently when calling `percyScreenshot` command the tasks were getting deleted as we used to find them using snapshot name.
- Since in screenshot command we upload comparisons they are allowed to have same snapshot name so now deletion cannot work, so we need to add find with tag name.

### CODE REVIEW

**Review by chinmay-browserstack — APPROVED** [review]

File: packages/core/src/utils.js [hunk 1]
```diff
@@ -512,3 +512,19 @@ export class DefaultMap extends Map {
     return super.get(key);
   };
 };
+
+export function compareObjectTypes(obj1, obj2) {
+  if (obj1 === obj2) return true; // Handles primitives
```
Comment (this-is-shivamsingh) [inline packages/core/src/utils.js:517]: - Do we need this, as this only check obj1 and obj2 reference and not the actual value.
- Below we are already doing this check `key in key2`

Comment (Amit3200) [reply packages/core/src/utils.js:517]: above is early exit for the primitive check as we don't want to spend time in converting data type and checking. Check lodash implementation may be.

File: packages/core/src/utils.js [hunk 2]
```diff
@@ -512,3 +512,19 @@ export class DefaultMap extends Map {
...
+  if (typeof obj1 !== 'object' || typeof obj2 !== 'object' || obj1 === null || obj2 === null) return false;
+
+  const keys1 = Object.keys(obj1);
+  const keys2 = Object.keys(obj2);
+
+  if (keys1.length !== keys2.length) return false;
+
+  for (const key of keys1) {
+    if (!keys2.includes(key) || !compareObjectTypes(obj1[key], obj2[key])) return false;
+  }
```
Comment (this-is-shivamsingh) [inline packages/core/src/utils.js:527]:
```
  name: tag.name || null,
                width: tag.width || null,
                height: tag.height || null,
                'os-name': tag.osName || null,
                'os-version': tag.osVersion || null,
                orientation: tag.orientation || null,
                'browser-name': tag.browserName || null,
                'browser-version': tag.browserVersion || null,
                resolution: tag.resolution || null
```
These are the only tag values, does it needes recursive checking ?

Comment (Amit3200) [reply packages/core/src/utils.js:527]: Tomorrow the tag can change and evolve, we need to ensure that cli is fail proof so that we don't miss cases.

Comment (this-is-shivamsingh) [reply packages/core/src/utils.js:527]: - Resolved it

**Review by pankaj443 — APPROVED** [review]

**Pull request merged**